File: bk-api/dao/settings_dao.go
package dao

import (
	"bk-api/entity"
	"database/sql"
)

type SettingsDAO interface {
	Save(tx *sql.Tx, settingEntity *entity.SettingEntity) (int64, error)
	FindByKey(key string) (*entity.SettingEntity, error)
	FindAll() (*[]entity.SettingEntity, error)
}

File: src/main/java/com/mageddo/bookmarks/dao/SettingsDAO.java
package com.mageddo.bookmarks.dao;

import java.util.List;

import com.mageddo.bookmarks.entity.SettingEntity;

public interface SettingsDAO {

  List<SettingEntity> findAll();

  SettingEntity find(String key);

  void patch(SettingEntity settingEntity);
}
